File: cpp/leetcode/796.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

class Solution
{
public:
    bool rotateString(string s, string goal)
    {
        int n = s.length();
        for (int i = 0; i < n; i++)
        {
            bool good = true;
            for (int j = 0; j < n; j++)
            {
                if (s[(i + j) % n] != goal[j])
                {
                    good = false;
                    break;
                }
            }
            if (good)
            {
                return true;
            }
        }
        return false;
    }
};

int main()
{
    Solution solution;
    cout << solution.rotateString("abcde", "cdeab") << "\n";
}

File: python/leetcode/796.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        
        def match(i):
            """Return True if the string s[i]... is equal to goal (considering s as circular string)"""
            if len(s) != len(goal):
                return False
            n = len(s)
            
            j = 0
            while j < n and s[(i+j)%n] == goal[j]:
                j += 1
            
            return j == n



        for i, ch in enumerate(s):
            if match(i):
                return True
            
        return False